contracts/vaults/VoltzVault.sol

```diff
@@ -278,167 +124,201 @@ contract VoltzVault is IVoltzVault, IntegrationVault {
+                    notional: (-liquidityDelta).toUint256(),
                     isMint: false,
                     marginDelta: 0
-                });
-
-                periphery.mintOrBurn(params);
+                }); 
             }
             // mint liquidity
             else {
-                IPeriphery.MintOrBurnParams memory params = IPeriphery.MintOrBurnParams({
+                params = IPeriphery.MintOrBurnParams({
                     marginEngine: _marginEngine, 
-                    tickLower: _currentPosition.low, 
-                    tickUpper: _currentPosition.high, 
-                    notional: opts.notionalForLiquidity.toUint256(),
+                    tickLower: _currentPosition.tickLower, 
+                    tickUpper: _currentPosition.tickUpper, 
+                    notional: liquidityDelta.toUint256(),
                     isMint: true,
                     marginDelta: 0
                 });
-                periphery.mintOrBurn(params);
             }
+
+            periphery.mintOrBurn(params);
+            _currentPositionLiquidity = (_currentPositionLiquidity.toInt256() + liquidityDelta).toUint256();
         }
+    }
 
-        // trade
-        if (opts.notionalForTrade != 0) {
-            uint160 sqrtPriceLimitX96 = opts.sqrtPriceLimitX96;
-            if (sqrtPriceLimitX96 == 0) {
-                if (opts.notionalForTrade < 0) {
-                    sqrtPriceLimitX96 = TickMath.MAX_SQRT_RATIO - 1;
-                } else {
-                    sqrtPriceLimitX96 = TickMath.MIN_SQRT_RATIO + 1;
-                }
-            }
+    function _updateCurrentPosition(TickRange memory ticks) internal {
+        Tick.checkTicks(ticks.tickLower, ticks.tickUpper);
+        require(ticks.tickLower % _tickSpacing == 0, ExceptionsLibrary.INVALID_VALUE);
+        require(ticks.tickUpper % _tickSpacing == 0, ExceptionsLibrary.INVALID_VALUE);
 
-            IPeriphery.SwapPeripheryParams memory params = IPeriphery.SwapPeripheryParams({
-                marginEngine: _marginEngine,
-                isFT: (opts.notionalForTrade < 0),
-                notional: (opts.notionalForTrade < 0) ? (-opts.notionalForTrade).toUint256() : opts.notionalForTrade.toUint256(),
-                sqrtPriceLimitX96: sqrtPriceLimitX96,
-                tickLower: _currentPosition.low,
-                tickUpper: _currentPosition.high,
-                marginDelta: 0
+        _currentPosition = ticks;
+    }
+
+    /// @notice Divide a given time in seconds by the number of seconds in a year
+    /// @param timeInSecondsAsWad A time in seconds in Wad (i.e. scaled up by 10^18)
+    /// @return timeInYearsWad An annualised factor of timeInSeconds, also in Wad
+    function _accrualFact(uint256 timeInSecondsAsWad)
+        internal
+        pure
+        returns (uint256 timeInYearsWad)
+    {
+        timeInYearsWad = timeInSecondsAsWad.div(SECONDS_IN_YEAR_IN_WAD);
+    }
+    
+    function _unwindAndExitCurrentPosition() internal {
```

Comment:
this function feels like it can be simplified, wdyt @0xZenus @avniculae ?

Reply:
Posted some thoughts below about this.